src/components/molecules/Visualizer/Engine/Cesium/core/Indicator.tsx

```diff
@@ -0,0 +1,133 @@
+  const transiton = useTransition(!!pos && isVisible, 500, {
+    mountOnEnter: true,
+    unmountOnExit: true,
+  });
+  const { indicator_type, indicator_image, indicator_img_scale } = property?.indicator ?? {};
+  const [img, w, h] = useIcon({ image: indicator_image, imageSize: indicator_img_scale });
+
+  useEffect(() => {
+    if (!viewer) return;
+    const handleTick = () => {
+      if (viewer.isDestroyed()) return;
+      const selected = viewer.selectedEntity;
+      if (
+        !selected ||
+        !selected.isShowing ||
+        !selected.isAvailable(viewer.clock.currentTime) ||
+        !selected.position
+      ) {
+        setIsVisible(false);
+        return;
+      }
+
+      // https://github.com/CesiumGS/cesium/blob/1.94/Source/Widgets/Viewer/Viewer.js#L1839
+      let position: Cartesian3 | undefined = undefined;
+      const boundingSphere = new BoundingSphere();
+      const state = (viewer.dataSourceDisplay as any).getBoundingSphere(
+        selected,
+        true,
+        boundingSphere,
+      );
+      // https://github.com/CesiumGS/cesium/blob/main/Source/DataSources/BoundingSphereState.js#L24
+      if (state !== 2 /* BoundingSphereState.FAILED */) {
+        position = boundingSphere.center;
+      } else if (selected.position) {
+        position = selected.position.getValue(viewer.clock.currentTime, position);
+      }
+
+      if (position) {
+        const pos = SceneTransforms.wgs84ToWindowCoordinates(viewer.scene, position);
+        setPos(pos);
+        setIsVisible(true);
+      } else {
+        setIsVisible(false);
+      }
+    };
+
+    viewer.clock.onTick.addEventListener(handleTick);
+    return () => {
+      if (viewer.isDestroyed()) return;
+      viewer.clock.onTick.removeEventListener(handleTick);
+    };
+  }, [viewer]);
+  // console.log( indicator_type );
+  return transiton !== "unmounted" && pos ? (
+    indicator_type === "crosshair" ? (
+      <StyledIcon
+        icon="crosshair"
+        className={className}
+        transition={transiton}
+        style={{ left: pos.x + "px", top: pos.y + "px" }}
+      />
+    ) : indicator_type === "default" ? (
+      <I
+        className={className}
+        transition={transiton}
+        style={{ left: pos.x + "px", top: pos.y + "px" }}
+      />
+    ) : (
+      <Image
+        src={img}
+        width={w}
+        height={h}
+        transition={transiton}
+        style={{ left: pos.x + "px", top: pos.y + "px" }}
+      />
+    )
+  ) : null;
+}
+
+const I = styled.div<{ transition: TransitionStatus }>`
+  position: absolute;
+  width: 50px;
+  height: 50px;
+  transform: translate(-50%, -50%);
+  transition: ${({ transition }) =>
+    transition === "entering" || transition === "exiting" ? "all 0.5s ease" : ""};
+  opacity: ${({ transition }) => (transition === "entering" || transition === "entered" ? 1 : 0)};
+`;
+const StyledIcon = styled(Icon)<{ transition: TransitionStatus }>`
+  position: absolute;
+  width: 50px;
+  height: 50px;
```

Comment:
Does the image scale correctly with this? Usually with the Icon component you need to pass it a `size` prop and the component handles the scaling. This is especially important with SVGs. Please make sure the StyledIcon is actually scaling correctly (in this case 50X50) and not just assuming it is, because it might be smaller/bigger than it should be.

Reply:
@KaWaite  please check is that what you mean ?

Reply:
@KaWaite  agreed on this right?

Reply:
@KaWaite  can I resolve this one